src/app/modules/main/wallet_section/filter.nim

```diff
@@ -0,0 +1,44 @@
+import strformat, sequtils, sugar
+
+import ./controller
+
+
+type Filter* = ref object
+  controller: Controller
+  addresses*: seq[string]
+  chainIds*: seq[int]
+
+proc initFilter*(
+  controller: Controller,
+): Filter =
+  result = Filter()
+  result.controller = controller
+  result.addresses = @[]
+  result.chainIds = @[]
+
+
+proc `$`*(self: Filter): string =
+  result = fmt"""WalletFilter(
+    addresses: {self.addresses},
+    chainIds: {self.chainIds},
+    )"""
+
+
+proc load*(self: Filter) =
+  let accounts = self.controller.getWalletAccounts()
+  self.addresses = @[accounts[0].address]
+  self.chainIds = self.controller.getNetworks().filter(n => n.enabled).map(n => n.chainId)
```

Comment:
@alaibe could you add this to Controller (or maybe to service):
```
proc getIdsOfEnabledNetworks*(self: Controller): seq[int] =
  return self.networkService.getNetworks().filter(n => n.enabled).map(n => n.chainId)

```

and use it here:
`self.chainIds = self.controller.getIdsOfEnabledNetworks()`

Reply:
will do